_includes/markdown/JavaScript.md

```diff
@@ -155,6 +155,74 @@ On all new projects you should be using up to date JavaScript methodologies comb
 
 Some older projects that have not yet been upgraded may not have the capability to use the most modern techniques, but it is still important to have processes in place that allow us to grow the technology stack as a project matures. In these cases, you should still follow best practice recommendations even if the newest patterns are not yet available to you.
 
+### Secure Your Code
+
+In JavaScript, we often have to insert new elements with dynamic attributes and content into the DOM. A common way to do this is to use the ```innerHTML``` method like so:
+
+```javascript
+const myElement = document.getElementById( 'myElement' );
+
+myElement.innerHTML = `<div class="container"><a href="${ someUrl }">${ someContent }</a></div>`;
+```
+
+However, passing HTML strings to ```innerHTML``` and methods like it (including jQuery's commonly used ```append``` and ```html``` methods) can expose your code to cross-site scripting, also known as XSS—the most common security vulnerability in JavaScript. Because these methods evaluate strings passed to them as HTML, they can execute potentially harmful code. For example, what happens if ```someContent``` in the above example is ```<img src="fakeImage" onerror="alert( 'hacked' )" />```? The JavaScript in the ```onerror``` attribute is executed.
+
+There are several measures you can take to circumvent this XSS vulnerability:
+
+#### Use ```textContent``` instead of ```innerHTML```
+
+When setting the human-readable content of a single element, using ```textContent``` is safer than using ```innerHTML``` because it does not parse strings as HTML—meaning any malicious code passed to it will not be executed.
+
+```javascript
+const myElement = document.getElementById( 'myElement' );
+
+myElement.textContent = 'This will get displayed as human-readable text, even if it contains HTML-like <syntax>.'
+```
+
+#### Use the ```Element``` API instead of HTML-parsing methods
+
+When you need to create multiple DOM elements, use JavaScript's [```Element``` API](https://developer.mozilla.org/en-US/docs/Web/API/Element) to create elements, set attributes and append them to the document. Creating your own elements and attributes will ensure that only those you explicitly define will make their way into the DOM. Refactoring our first example:
+
+```javascript
+const myElement = document.getElementById( 'myElement' );
+
+// Create the new elements.
+const container = document.createElement( 'div' );
+const anchor = document.createElement( 'a' );
+
+// Set attributes and content.
+container.setAttribute( 'class', 'container' );
+anchor.setAttribute( 'href', someUrl );
+anchor.textContent = someContent;
```

Comment:
Put someContent in single quote.

Reply:
I assigned it a value instead in all examples.